lib/WUI/nhttp/gcode_upload.cpp

```diff
@@ -189,6 +219,39 @@ Step GcodeUpload::step(string_view input, bool terminated_by_client, uint8_t *,
     }
 }
 
+Step GcodeUpload::stepPUT(string_view input) {
+    const size_t read = std::min(input.size(), size_rest);
+    // remove the "/usb/" prefix
+    const char* filename = filepath + USB_MOUNT_POINT_LENGTH;
+
+    // check only once
+    static bool filename_checked = false;
```

Comment:
So, check only on the first ever file that gets uploaded to the printer during its runtime? :-O

What exactly are you trying to solve here? The fact that the step might be called multiple times? Don't you want to put the check at the beginning, somewhere in the `start` static method, which is full of other one-time checks?

If at least possible, avoid any kind of "globals" unless they are conceptually needed.